### PR TITLE
Create zh-trans:set-up-coredns-provider-federation.md

### PR DESCRIPTION
**Pull request by zhangx501**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Remember to delete this note before submitting your pull request.
>
> For pull requests on 1.15 Features: set Milestone to 1.15 and Base Branch to dev-1.15
> 
> For pull requests on Chinese localization, set Base Branch to release-1.14
>
> For pull requests on Korean Localization: set Base Branch to dev-1.14-ko.\<latest team milestone>
>
> If you need Help on editing and submitting pull requests, visit:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> If you need Help on choosing which branch to use, visit:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

**Comment by zhangx501** [comment]
/assign @dchen1107
Please give this PR approval.

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+{{< deprecationfilewarning >}}
+{{< include "federation-deprecation-warning-note.md" >}}
+{{< /deprecationfilewarning >}}
+
+<!--
+This page shows how to configure and deploy CoreDNS to be used as the
+DNS provider for Cluster Federation.
+-->
+本页显示如何配置和部署 CoreDNS ，用作 Cluster Federation 的 DNS 提供端。
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]: Cluster Federation -> 联邦集群 [1]
提供端 -> 提供者 [2]

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+<!--
+* You need to have a running Kubernetes cluster (which is
+referenced as host cluster). Please see one of the
+[getting started](/docs/setup/) guides for
+installation instructions for your platform.
+* Support for `LoadBalancer` services in member clusters of federation is
+mandatory to enable `CoreDNS` for service discovery across federated clusters.
+-->
+*  你需要一个正在运行的 Kubernetes 集群（它被引用为主机集群）。请参阅其中的一个[入门](/docs/setup/) 指，来为你的平台进行安装。
+*  联合成员集群中对`LoadBalancer`服务的支持是必须启用`CoreDNS`以跨联合集群进行发现服务。
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]:
```suggestion
*  联邦成员集群中对 `LoadBalancer` 服务的支持是必须启用 `CoreDNS` 以跨联邦集群进行服务发现。
```

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+-->
+CoreDNS 可以部署在各种配置中。以下解释是一个参考，可以调整以满足平台和集群联盟的需要。
+
+<!--
+To deploy CoreDNS, we shall make use of helm charts. CoreDNS will be
+deployed with [etcd](https://coreos.com/etcd) as the backend and should
+be pre-installed. etcd can also be deployed using helm charts. Shown
+below are the instructions to deploy etcd.
+-->
+要部署 CoreDNS， 我们将使用 helm 图表。CoreDNS 将与[etcd](https://coreos.com/etcd)一起部署，作为后端，应该预先安装。etcd 也可以用 helm 图表来部署。以下显示的是部署 etcd 的说明。
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]:
```suggestion
要部署 CoreDNS， 我们将使用 helm 图表。CoreDNS 将与 [etcd](https://coreos.com/etcd) 一起部署，作为后端，应该预先安装。etcd 也可以用 helm 图表来部署。以下显示的是部署 etcd 的说明。
```

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+    helm install --namespace my-namespace --name etcd-operator stable/etcd-operator
+    helm upgrade --namespace my-namespace --set cluster.enabled=true etcd-operator stable/etcd-operator
+    
+<!--
+*Note: etcd default deployment configurations can be overridden, suiting the
+host cluster.*
+-->
+*说明: etcd 默认部署可以被覆盖，适合主机群。*
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]: 主机群 -> 主机集群

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+<!--
+After deployment succeeds, etcd can be accessed with the
+[http://etcd-cluster.my-namespace:2379](http://etcd-cluster.my-namespace:2379) endpoint within the host cluster.
+-->
+部署成功后， etcd 可以用[http://etcd-cluster.my-namespace:2379](http://etcd-cluster.my-namespace:2379) 访问，是主集群中的终结点。
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]:
```suggestion
部署成功后， etcd 可以用 [http://etcd-cluster.my-namespace:2379](http://etcd-cluster.my-namespace:2379) 访问，是主机集群中的端点。
```

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+<!--
+The CoreDNS default configuration should be customized to suit the federation.
+Shown below is the Values.yaml, which overrides the default
+configuration parameters on the CoreDNS chart.
+-->
+应自定义 CoreDNS 默认配置以适应联盟。下面显示的是 Values.yaml，它会覆盖 CoreDNS 图表上的默认设置参数。
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]: [3]

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+ - Enable `plugins.etcd` by setting `plugins.etcd.enabled` to
+true.
+ - Configure the DNS zone (federation domain) for which CoreDNS is
+authoritative by setting `plugins.etcd.zones` as shown above.
+ - Configure the etcd endpoint which was deployed earlier by setting
+`plugins.etcd.endpoint`
+-->
+以上的配置文件需要一些解释：
+
+ - `isClusterService` 指定是否应将 CoreDNS 部署为 cluster-service，这是默认设置。您应该将它设置为false，这样 CoreDNS 作为一个 Kubernetes 应用服务被部署。
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]:
```suggestion
 - `isClusterService` 指定是否应将 CoreDNS 部署为 cluster-service，这是默认设置。您应该将它设置为 false，这样 CoreDNS 作为一个 Kubernetes 应用服务被部署。
```

File: content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+ - `serviceType`为 CoreDNS 指定要创建的 Kubernetes 服务类型。你需要选择"LoadBalancer" 或 "NodePort" 使 Kubernetes 集群外的 CoreDNS 服务可访问。
```
Comment (xichengliudui) [inline content/zh/docs/tasks/federation/zh-trans:set-up-coredns-provider-federation.md]:
```suggestion
 - `serviceType` 为 CoreDNS 指定要创建的 Kubernetes 服务类型。你需要选择 `LoadBalancer` 或 `NodePort` 使 Kubernetes 集群外的 CoreDNS 服务可访问。
```

**Comment by xichengliudui** [comment]
A sad news, i just found out that this article has been translated by me.
And it has merged.
https://github.com/kubernetes/website/pull/15902

**Comment by xichengliudui** [comment]
And your file name is wrong.

**Pull request closed without merging**